library/ns1_record.py

```diff
@@ -450,17 +493,71 @@ def update(self, zone, record):
                     changed = True
                     args[key] = input_data
 
+        # create a new copy of the previously sanitized dict that will be updated with chaning args to support --diff
+        after_changes = copy.deepcopy(record_data)
+        for k, v in args.items():
+            if after_changes[k] != v:
+              after_changes[k] = v
```

Comment:
this isn't quite the same situation as in the zone diff PR, but I think it might be close enough that we could (re)use the more nuanced change-merging logic I'm proposing. We're not mutating what goes to the API here, but we could still maybe get better diffs